test/test_codegen.py

```diff
@@ -128,26 +88,84 @@ def test_get_pressure(mechname):
     x[i_fu] = (alpha * phi) / (nu + alpha * phi)
     x[i_ox] = nu * x[i_fu] / phi
     x[i_di] = (1.0 - alpha) * x[i_ox] / alpha
+
     # Get equilibrium composition
     sol.TPX = t, ct.one_atm, x
     sol.equilibrate("UV")
     t, rho, y = sol.TDY
     p_ct = sol.P
+
     # Compute pressure with pyrometheus and compare to Cantera
     p_pm = ptk.get_pressure(rho, t, y)
-    assert abs(p_ct - p_pm) / p_ct < 1.0e-2
+    assert abs(p_ct - p_pm) / p_ct < 1.0e-12
+
+
+@pytest.mark.parametrize("mechname", ["uiuc", "sanDiego"])
+def test_get_thermo_properties(mechname):
+    """This function tests that pyrometheus-generated code
+    computes thermodynamic properties c_p, s_r, h_rt, and k_eq
+    correctly by comparing against Cantera"""
+    # Create Cantera and pyrometheus objects
+    sol = ct.Solution(f"mechs/{mechname}.cti", "gas")
+    ptk = pyro.get_thermochem_class(sol)()
+
+    # Loop over temperatures
+    temp = np.linspace(500.0, 3000.0, 10)
+    for t in temp:
+
+        # Set state in cantera for comparison
+        sol.TP = t, ct.one_atm
+
+        # Get properties from pyrometheus and compare to Cantera
+        cp_pm = ptk.get_species_specific_heats_r(t)
+        cp_err = np.linalg.norm(cp_pm - sol.standard_cp_R, np.inf)
+        print(f"cp_pm = {cp_pm}")
+        print(f"cnt_cp = {sol.standard_cp_R}")
+        assert cp_err < 1.0e-13
+
+        s_pm = ptk.get_species_entropies_r(t)
+        s_err = np.linalg.norm(s_pm - sol.standard_entropies_R, np.inf)
+        print(f"s_pm = {s_pm}")
+        print(f"cnt_s = {sol.standard_entropies_R}")
+        assert s_err < 1.0e-13
+
+        h_pm = ptk.get_species_enthalpies_rt(t)
+        h_err = np.linalg.norm(h_pm - sol.standard_enthalpies_RT, np.inf)
+        print(f"h_pm = {h_pm}")
+        print(f"cnt_h = {sol.standard_enthalpies_RT}")
+        assert h_err < 1.0e-13
+
+        keq_pm1 = ptk.get_equilibrium_constants(t)
+        print(f"keq1 = {keq_pm1}")
+        keq_pm = 1.0 / np.exp(ptk.get_equilibrium_constants(t))
+        keq_ct = sol.equilibrium_constants
+
+        print(f"keq_pm = {keq_pm}")
+        print(f"keq_cnt = {keq_ct}")
+        print(f"temperature = {t}")
+        # Exclude check on irreversible reactions
```

Comment:
Why? Maybe add some context on why that's difficult.

Reply:
Pyro doesn't "do" irreversible reactions.  Help me out @ecisneros8 ?

Reply:
Pyro does do irreversible reactions. We even compute their equilibrium constants, even though they are meaningless and never factor into the computation of the rates, to maintain the equilibrium-constants array at size N_reactions. However, we don't compare them against Cantera because, well, they are meaningless.

Reply:
So my current plan would be to replace the comment with:
`# Exclude meaningless check on equilibrium constants for irreversible reactions`

Reply:
```suggestion
        # Exclude meaningless check on equilibrium constants for irreversible reactions
```